Add docs for batch operations (#17283)
ydb-platform/ydb: ydb/docs/en/core/yql/reference/syntax/batch-delete.md

# BATCH DELETE

{% note tip %}

Before diving into `BATCH DELETE`, it is recommended to familiarize yourself with the standard [DELETE FROM](delete.md).

{% endnote %}

> `BATCH DELETE` allows to delete records in large tables while minimizing the risk of lock invalidation and transaction rollback by weakening guarantees. Specifically, data deletion is performed as a series of transactions for each [partition](../../../concepts/datamodel/table.md#partitioning) of the specified table separately, processing 10 000 rows per iteration. At the time of request execution, 10 partitions are processed simultaneously.

Reviewer: ```suggestion
`BATCH DELETE` allows to delete records in large tables while minimizing the risk of lock invalidation and transaction rollback by weakening guarantees. Specifically, data deletion is performed as a series of transactions for each [partition](../../../concepts/datamodel/table.md#partitioning) of the specified table separately, processing 10 000 rows per iteration. Each query processes up to 10 partitions concurrently.
```

> `BATCH DELETE` allows to delete records in large tables, minimizing the risk of lock invalidation and transaction rollback by weakening guarantees, namely, data deletion is performed as a series of transactions for each [partition](../../../concepts/datamodel/table.md#partitioning) of the specified table separately, processing a limited number of rows per iteration (by default, 10000 rows). At the time of request execution, a limited number of partitions are processed simultaneously (by default, no more than 10 partitions).

Reviewer: ```suggestion
`BATCH DELETE` allows to delete records in large tables while minimizing the risk of lock invalidation and transaction rollback by weakening guarantees. Specifically, data deletion is performed as a series of transactions for each [partition](../../../concepts/datamodel/table.md#partitioning) of the specified table separately, processing a limited number of rows per iteration (by default, 10000 rows). At the time of request execution, a limited number of partitions are processed simultaneously (by default, no more than 10 partitions).
```

> `BATCH DELETE` allows to delete records in large tables, minimizing the risk of lock invalidation and transaction rollback by weakening guarantees, namely, data deletion is performed as a series of transactions for each [partition](../../../concepts/datamodel/table.md#partitioning) of the specified table separately, processing a limited number of rows per iteration (by default, 10000 rows). At the time of request execution, a limited number of partitions are processed simultaneously (by default, no more than 10 partitions).

Reviewer: How to change these defaults?

Author: > Our team is unavailable to make a new documentation for the whole configuration right now, so I'd like to add this part for BATCH queries and skip the rest.

UPD: KQP will create a ticket for the description of table service config, for now all references to default values ​​have been removed.

This query, like the standard `DELETE`, is executed synchronously and completes with some status. If an error occurs or the client is disconnected, the data delete stops, and the applied changes are not rolled back.

The semantics are inherited from the standard `DELETE` with the following restrictions:

* Supported only for [row-oriented tables](../../../concepts/glossary.md#row-oriented-table).

> * Supported only for transactions in the `NoTx` mode.

Reviewer: What's the `NoTx` mode? Add link to explanation/examples.

* The use of subqueries and multiple queries in a single expression, including `DELETE FROM ... ON`, is prohibited.

> * The `RETURNING` keyword is unavailable.

Reviewer: ```suggestion
* The `RETURNING` clause is unavailable.
```
